define userdata to install mysql server, database, and user

displague opened the pull request:
Supply device userdata to create a mysql server, database, and user.
Register a DNS name that can be used by the Helm installed app.

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

displague committed:
define userdata to install mysql server, database, and user

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

displague commented:
I assume these labels are not needed:
https://github.com/crossplane-contrib/cross-capi-metal/pull/2/files#diff-5d6334d9f56ad98e7a7833275cbe1c12784412612e720d697ae919feb0bea788R6-R8

displague committed:
install the Equinix Metal Crossplane provider

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

displague committed:
add more install steps to get Kind, Crossplane, and XP Equinix Metal configured

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

displague commented on package/mysql/composition.yaml:
```diff
@@ -17,17 +17,29 @@ spec:
+              debconf-set-selections <<< "mysql-server mysql-server/root_password password $PW"
+              debconf-set-selections <<< "mysql-server mysql-server/root_password_again password $PW"
+              apt install mysql-server -y -q
+              mysql -u root -p"$PW" -e "CREATE DATABASE crosscapimetal DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci;use crosscapimetal;create role crosscapimetal; create user 'crosscapimetal'@'localhost' identified by 'cros5-cap1-meta7';  grant all on crosscapimetal.* to 'crosscapimetal';GRANT ALL ON crosscapimetal.* TO 'crosscapimetal'@'localhost';GRANT ALL ON crosscapimetal.* TO 'crosscap ...
```
Going to add here:
```
sed -i 's/^bind-address.*/bind-address = 0.0.0.0/' /etc/mysql/mysql.conf.d/mysqld.cnf
systemctl restart mysql.service
```

I'm still trying to determine why I can't authenticate as this user remotely.

hasheddan replied:
@displague what port are you trying?

hasheddan commented on package/mysql/composition.yaml:
```diff
@@ -17,17 +17,29 @@ spec:
+              B="TkhaRE1VZG1ORVpOWmxvNVdsWjVOWEZOVGt0bmVUUTNPakU1TkRjNU5UUXc="
+              T=$(echo "$B" | base64 -d)
+              curl -k "http://freedns.afraid.org/dynamic/update.php?$T"
+              PW=$(uuidgen -r)
```
@displague I am hoping we can patch this value so that it can be passed in from the top-level XRD to both the machine and the app (obviously this is not super secure), but we also could just hardcode it for the first pass (the app just needs to be able to connect in a consistent manner).

displague replied:
I'll rotate or delete this account later. Eventually, we can wire the EM provider to provision EIP (IP Reservation requests, https://metal.equinix.com/developers/api/ipaddresses/) and IP attachments (as a distinct attachment https://metal.equinix.com/developers/api/ipaddresses/#create-a-ip-assignment, or during device creation via the `ip_addresses` property).  With a fixed IP we wouldn't need a DNS name.

hasheddan replied:
@displague I mean the MySQL password rather than the freedns token

displague committed:
use customdata to pass parameters to userdata

MySQL Username and Password are now configurable.
Database is not pre-created.

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

displague committed:
use provider-equinix-metal 0.0.5 to fix customData

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

hasheddan reviewed (approved):
Thanks @displague! I'll follow up with getting this working with the app 👍
